src/Adapter/Product/CommandHandler/DeleteAllProductCategoriesHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductCategoryHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\DeleteAllProductCategoriesCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\DeleteAllProductCategoriesHandlerInterface;
+
+/**
+ * Handles @see DeleteAllProductCategoriesCommand using legacy object model
+ */
+final class DeleteAllProductCategoriesHandler extends AbstractProductCategoryHandler implements DeleteAllProductCategoriesHandlerInterface
+{
+    /**
+     * @var int
+     */
+    private $homeCategoryId;
+
+    /**
+     * @param int $homeCategoryId
+     */
+    public function __construct(int $homeCategoryId)
+    {
+        $this->homeCategoryId = $homeCategoryId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(DeleteAllProductCategoriesCommand $command): void
+    {
+        $product = $this->getProduct($command->getProductId());
+
+        $this->updateCategories($product, [$this->homeCategoryId]);
```

Comment:
Does a product always need to be associated with a category?

Reply:
yes, according to products specification `A category is necessarily selected, it is impossible not to have one.` and current product page behavior - does not allow to unselect last category.